### PR TITLE
Integration specs

### PR DESCRIPTION
**Pull request by 2bleO**
**In this PR:**
1. Add integration tests for Login and User index pages
2. Add integration tests for User show page
3. Add integration tests for Post index page
4. Add integration tests for Post show page

### CODE REVIEW

File: spec/views/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    it 'filling email and password fields with correct data redirects to the root page' do
+      fill_in 'Email', with: 'test@user.com'
+      fill_in 'Password', with: 'testu1'
+      click_button 'Log in'
+      expect(page).to have_current_path(home_path)
+    end
+  end
+end
+
+RSpec.describe 'list of users in', type: :feature do
```
Comment (Davidosky007) [inline spec/views/user_index_spec.rb:45]: * Kindly add the `Tests that asserts you can see the profile picture for each user ` as it is required in the project requirements.

File: spec/views/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe 'User view', type: :feature do
```
Comment (Davidosky007) [inline spec/views/user_show_spec.rb:6]: * Kindly add the following missing tests as they are required 
* `Tests that asserts you can see the user's profile picture.`
* `Tests that asserts you can see the user's first 3 posts.`
* `Tests that redirects to the post's show page when you click on a user's post.`

File: spec/views/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+RSpec.describe 'Post view', type: :feature do
```
Comment (Davidosky007) [inline spec/views/post_index_spec.rb:6]: * I suggest you add these missing tests as they are required 
* ` Tests that asserts you can see the user's profile picture.`

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @2bleO 👋, 

Good job so far! 💪 
* No linter errors ✔️ 
* Descriptive commit messages ✔️ 
* Well written PR ✔️ 
* Proper use of git-flow ✔️ 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
* I suggest you fix this error i got while running `rails db:seed`  which as a result causes some tests to fail kindly fix these errors 🙏 
image
image

 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @2bleO

Great work on making the project requirements 👏 
You've done well implementing some of the project requirements, but there are still some which aren't addressed yet, please work on it and ask for another review.


 ## Required Changes ♻️
 
_Check the comments under the review._
As the previous reviewer mentioned above `rails db:seed` doesn't work and all the tests are not passing successfully. 

delete-1

delete


## Optional suggestions

- N/A 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Comment by 2bleO** [comment]
Hello @Davidosky007 & @Emmyn5600 can you guys please help me on how to solve this? Because on my side, I'm not having that error. db:seed and/or db:setup works perfect on my side and all test pass successfully. I can't solve this issue if I can not replicate it on my side.

Here a Loom video link for you to see how it works on my side: https://www.loom.com/share/247d733f19174b039fb7473af2f26626

**Comment by Davidosky007** [comment]
> Hello @Davidosky007 & @Emmyn5600 can you guys please help me on how to solve this? Because on my side, I'm not having that error. db:seed and/or db:setup works perfect on my side and all test pass successfully. I can't solve this issue if I can not replicate it on my side.
> 
> Here a Loom video link for you to see how it works on my side: https://www.loom.com/share/247d733f19174b039fb7473af2f26626

Hi, future code reviewer in case you encounter this error ` Rails: Webpacker::Manifest::MissingEntryError in Home#index` reported by the previous reviewer kindly run this code `bundle exec rails webpacker:install ` 🙏

**Comment by rloterh** [comment]
> > Hello @Davidosky007 & @Emmyn5600 can you guys please help me on how to solve this? Because on my side, I'm not having that error. db:seed and/or db:setup works perfect on my side and all test pass successfully. I can't solve this issue if I can not replicate it on my side.
> > Here a Loom video link for you to see how it works on my side: https://www.loom.com/share/247d733f19174b039fb7473af2f26626
> 
> Hi, future code reviewer in case you encounter this error ` Rails: Webpacker::Manifest::MissingEntryError in Home#index` reported by the previous reviewer kindly run this code `bundle exec rails webpacker:install ` 🙏

@Davidosky007 
Kindly use  `bundle exec rails webpacker:install ` cautiously, as this might make other features not work, especially with React associated projects

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Onesis, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have no more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**